README.md

```diff
@@ -0,0 +1,767 @@
+[<img width="250" alt="ImageKit.io" src="https://raw.githubusercontent.com/imagekit-developer/imagekit-javascript/master/assets/imagekit-light-logo.svg"/>](https://imagekit.io)
+
+# ImageKit.io Next.js SDK
+
+[![Node CI](https://github.com/imagekit-developer/imagekit-next/workflows/Node%20CI/badge.svg)](https://github.com/imagekit-developer/imagekit-next/)
+[![npm version](https://img.shields.io/npm/v/imagekit-next)](https://www.npmjs.com/package/imagekit-next)
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+[![Twitter Follow](https://img.shields.io/twitter/follow/imagekitio?label=Follow&style=social)](https://twitter.com/ImagekitIo)
+
+ImageKit Next.js SDK allows you to resize, optimize, deliver, and upload images and videos in your Next.js application.
+
+ImageKit is a complete media storage, optimization, and transformation solution that comes with an image and video CDN. It can be integrated with your existing infrastructure - storage like AWS S3, web servers, your CDN, and custom domain names, allowing you to deliver optimized images in minutes with minimal code changes.
+
+## Installation
+
+```shell
+npm install --save imagekit-next
+```
+
+or
+
+```shell
+yarn add imagekit-next
+```
+
+## Usage
+
+### Initialization
+
+Import components in your code:
+
+```js
+import { IKImage, IKVideo, ImageKitProvider, IKUpload, ImageKitContext } from 'imagekit-next'
+```
+
+### Pages Router (/src)
+
+By default, `imagekit-next` fully supports the included components in the `src` directory without any additional setup.
+
+### App Router (/app)
+
+To use `imagekit-next` in the app directory, you currently need to mark the parent page or component as a Client Component.
+
+This requirement exists because components are divided into two types: Client Components and Server Components.
+
+To do this, simply add the following at the top of the file:
+
+```js
+"use client";
+```
+
```

Comment:
### App Router (/app)

To use `imagekit-next` in the app directory, you currently need to mark the parent page or component as a Client Component.

This requirement exists because components are divided into two types: Client Components and Server Components.

To do this, simply add the following at the top of the file:

```js
"use client";
```
The components in the app directory are by default React Server Side Components (RSCs), but not everything can be done server-side. That's where Client components come in.

`imagekit-next` provides components that operate exclusively on client side.

When using the sdk for the first time, you might enter the following error
```
Error: (0 , react__WEBPACK_IMPORTED_MODULE_0__.createContext) is not a function
```
or 
```
You're importing a component that needs createContext. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
```

To simply fix this, you can force the component or the page to operate solely on Client Side by adding `"use client";` directive at the start of the file, you can read more about it here [Using Client Components in Next.js](https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs)

Reply:
done